### PR TITLE
[Feature](avro) Support Apache Avro file format

### PR DESCRIPTION
**Pull request by DongLiang-0**
# Proposed changes

#18431

## Problem summary
support read avro file by hdfs() or s3() .

current avro reader only support common data type, the complex data types will be supported later

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by DongLiang-0** [comment]
run buildall

**Comment by DongLiang-0** [comment]
run buildall

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 35.42 seconds
 stream load tsv:          433 seconds loaded 74807831229 Bytes, about 164 MB/s
 stream load json:         24 seconds loaded 2358488459 Bytes, about 93 MB/s
 stream load orc:          60 seconds loaded 1101869774 Bytes, about 17 MB/s
 stream load parquet:          30 seconds loaded 861443392 Bytes, about 27 MB/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230516051549_clickbench_pr_144291.html

File: be/src/vec/exec/scan/avro_reader.cpp [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+            required_param.insert(std::make_pair("uri", params.hdfs_params.hdfs_conf.data()->value));
+            break;
+        case TFileType::FILE_S3:
+            required_param.insert(params.properties.begin(), params.properties.end());
+            break;
+        case TFileType::FILE_LOCAL:
+        case TFileType::FILE_BROKER:
+        case TFileType::FILE_STREAM:
+        default:
+            Status::InternalError("unsupported file reader type: {}", std::to_string(type));
```
[the remark this answers is not in the document]
Comment (DongLiang-0) [reply be/src/vec/exec/scan/avro_reader.cpp:92]: Thanks for your suggestion, I have done this.

File: fe/fe-core/src/main/java/org/apache/doris/tablefunction/S3TableValuedFunction.java [hunk 2]
```diff
@@ -79,12 +80,15 @@ public S3TableValuedFunction(Map<String, String> params) throws AnalysisExceptio
...
         }
 
-        parseProperties(tvfParams);
         // set S3 location properties
         // these five properties is necessary, no one can be lost.
         locationProperties = S3Properties.credentialToMap(credential);
         String usePathStyle = tvfParams.getOrDefault(PropertyConverter.USE_PATH_STYLE, "false");
         locationProperties.put(PropertyConverter.USE_PATH_STYLE, usePathStyle);
+        locationProperties.put(S3Properties.VIRTUAL_BUCKET, virtualBucket);
+        locationProperties.put(S3Properties.VIRTUAL_KEY, getVirtualKey());
```
[the remark this answers is not in the document]
Comment (DongLiang-0) [reply fe/fe-core/src/main/java/org/apache/doris/tablefunction/S3TableValuedFunction.java:89]: The original `BUCKET` value is not correct. I am not sure if this is a bug. There are quite a lot of places involved in this variable, so I did not change it and created a new `VIRTUAL_BUCKET`.

File: fe/java-udf/pom.xml [hunk 3]
```diff
@@ -45,6 +45,26 @@ under the License.
...
+        <dependency>
+            <groupId>org.apache.avro</groupId>
+            <artifactId>avro</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.hadoop</groupId>
+            <artifactId>hadoop-client</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.hadoop</groupId>
```
[the remark this answers is not in the document]
Comment (DongLiang-0) [reply fe/java-udf/pom.xml:59]: Thanks for your suggestion, I have done this.

**Pull request closed without merging**